package.json

```diff
@@ -819,6 +819,25 @@
           "default": false,
           "description": "%config.editor_prefer_editor%"
         },
+        "i18n-ally.keygenStrategy": {
+          "type": "string",
+          "default": "default",
+          "enum": [
+            "default",
+            "random"
+          ],
+          "description": "%config.keygen_strategy%"
+        },
+        "i18n-ally.keyPrefix": {
+          "type": "string",
+          "default": "",
+          "description": "%config.key_prefix%"
+        },
+        "i18n-ally.useBestMatchFile": {
+          "type": "boolean",
+          "default": false,
+          "description": "%config.use_best_match_file%"
+        },
```

Comment:
I don't really understand what this "best match" for. Can you have some examples for me?

Reply:
With previous way, if ally found more than one locale files , it will ask user to select which file to extract to. With this option enabled, ally will compare current extracting file's path to all locale files. Using npm package [string-similarity](https://www.npmjs.com/package/string-similarity), which is just compute the degree of similarity of paths. And automatically extract text to the most matched file.

For example, 
I extract text from src/aaa/bbb/ccc.vue

Locale files are 
locale/aaa/zh-CN.json
locale/aaa/bbb/zh-CN.json
locale/aaa/bbb/ccc/zh-CN.json

The best matched path will be `locale/aaa/bbb/ccc/zh-CN.json`. 

In a nutshell, this resolve https://github.com/antfu/i18n-ally/issues/271

Reply:
The key name is someway confusing, so I change it to `autoExtractToBestMatchedFile`

Reply:
Ok I see, that's an interesting approach. But I am thinking about the naming, emm.. maybe something like:
```jsonc
"i18n-ally.extract.targetPickingStrategy": "most-similar" // default to none
```
What do you think?

Reply:
OK. Change to targetPickingStrategy. And from boolean to string. Two option: 'none' and 'most-similar'.